interop/xds/client/client.go

```diff
@@ -39,18 +42,43 @@ type statsWatcherKey struct {
 	endID   int32
 }
 
+// rpcInfo contains the rpc type and the hostname where the response is received
+// from.
+type rpcInfo struct {
+	typ      rpcType
+	hostname string
+}
+
 type statsWatcher struct {
 	rpcsByPeer    map[string]int32
+	rpcsByType    map[rpcType]map[string]int32
 	numFailures   int32
 	remainingRpcs int32
-	c             chan *testpb.SimpleResponse
+	chanHosts     chan *rpcInfo
+}
+
+func (watcher *statsWatcher) buildResp() *testpb.LoadBalancerStatsResponse {
+	rpcsByType := make(map[string]*testpb.LoadBalancerStatsResponse_RpcsByPeer)
```

Comment:
Optional, but usually a good idea: `make(map[]..., len(watcher.rpcsByType))`

...or, would it be better to store `rpcsByType` as `map[type-string]RpcsByPeer` so it doesn't need to be re-formed at all?

Reply:
Done. Added length.
And removed `rpcType` all together.